src/types/events/index.ts

```diff
@@ -36,12 +18,18 @@ export interface SlackEventMiddlewareArgs<EventType extends string = string> {
   ack?: undefined;
 }
 
+interface BaseSlackEvent<T extends string = string> {
+  type: T;
+}
+export type EventTypePattern = string | RegExp;
+export type FunctionInputs = Record<string, unknown>;
+
 /**
  * A Slack Events API event wrapped in the standard envelope.
  *
  * This describes the entire JSON-encoded body of a request from Slack's Events API.
  */
-export interface EnvelopedEvent<Event = BasicSlackEvent> extends StringIndexed {
```

Comment:
@filmaj This export removal caused a breaking change for existing apps that rely on the type: https://github.com/slackapi/bolt-js/issues/2243 Can we revert this change at least for 3.x series?